src/treetensornetworks/opsum_to_ttn.jl

```diff
@@ -114,39 +143,50 @@ function ttn_svd(
   site_coef_done = Prod{Op}[] # list of terms for which the coefficient has been added to a site factor
   # temporary symbolic representation of TTN Hamiltonian
   tempTTN = Dict(v => QNArrElem{Scaled{coefficient_type,Prod{Op}},degrees[v]}[] for v in vs)
-
+  #ToDo: precompute span of each term and store
+  # compute span of each term
+  spans = Dict{eltype(os),Set{vertextype_sites}}()
+  for term in os
+    spans[term] = span(term, sites)
+  end
   # build compressed finite state machine representation
   for v in vs
     # for every vertex, find all edges that contain this vertex
-    edges = filter(e -> dst(e) == v || src(e) == v, es)
+    edges = ordered_incident_edges(sites, v, es)
+
     # use the corresponding ordering as index order for tensor elements at this site
     dim_in = findfirst(e -> dst(e) == v, edges)
     edge_in = (isnothing(dim_in) ? [] : edges[dim_in])
     dims_out = findall(e -> src(e) == v, edges)
     edges_out = edges[dims_out]
 
+    which_incident_edge = map_vertices_to_incident_edges(sites, v, edges)
+    factor_to_edge(t) = which_incident_edge[ITensors.site(t)]
```

Comment:
I find the name `factor_to_edge` confusing.

Reply:
We could just call it `which_incident_edge(t)`?

Reply:
I renamed it to `vertices_to_incident_edges_dict` and added some comments to explain what it's doing.